Adds an extra signaler check to compare frequencies with the incoming signal

Jackraxxus opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
So, signalers are meant to pulse when they receive a signal that matches their EXACT code and frequency. However, for some reason 145.7 or FREQ_SIGNALER bypasses this and pings everything with the same code regardless of the receiving signaler's frequency. I do not know why this happens, it just does. This PR adds an extra check to compare the incoming signal's frequency with the receiver's frequency to prevent this bug.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
One out of one people I surveyed didn't enjoy having their lab blown up randomly.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: The 145.7 frequency is no longer received by all signalers.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Jackraxxus committed:
Descriptive commit name so that if I forget to change the PR title again the commit works as a backup title

If you see this commit in the PR then it works, if you don't see this commit then it didn't work. Very strange how that happens.

Jackraxxus commented:
WASTED trips
Also it should be signaller, not signaler. I've mistyped that like 50 times in the making of this PR and I'm mad.

AnturK commented on code/modules/assembly/signaler.dm:
```diff
@@ -136,13 +136,15 @@
 	. = FALSE
 	if(!signal)
 		return
+	if(signal.data["freq"] != frequency)
```
From quick glance the issue stems more from signaler topic calls doing raw frequency changes instead of calling set_frequency. Signals already work on filtered listener lists.

Jackraxxus committed:
Fixes the bug but in a better way I think

I've tested it and it works. I'm not actually sure if this is what AnturK was asking for though it's 2 am and my thigh-highs are in the wash so my ability to speak coder is severely impaired.

AnturK commented on code/modules/assembly/signaler.dm:
```diff
@@ -98,9 +98,9 @@
-			frequency = unformat_frequency(params["freq"])
-			frequency = sanitize_frequency(frequency, TRUE)
-			set_frequency(frequency)
+			new_frequency = unformat_frequency(params["freq"])
+			new_frequency = sanitize_frequency(new_frequency, TRUE)
+			set_frequency(new_frequency)
```
```suggestion
			var/new_frequency = sanitize_frequency(unformat_frequency(params["freq"]), TRUE)
			set_frequency(new_frequency)
```
Should just be just local variable instead of adding new parameter.

AnturK replied:
Also from quick search, there's also a case of this with pressure plates, can you fix this as well ?

AnturK replied:
AAnd it's also copypasted in signaler program.

Jackraxxus replied:
Right, yeah. I'll see if I can patch those.

Jackraxxus committed:
Review commit

Co-authored-by: AnturK <AnturK@users.noreply.github.com>

Jackraxxus committed:
Makes it compile, applies chance to the modular computer app

Jackraxxus committed:
Also applies the change to pressure plates

I'm not sure I've ever seen one of these in game. Guess people aren't creative enough to really make use of them. Do they have circuit integration? Also triple posting in commits, very messy.
